From nobody Mon Sep 17 00:00:00 2001
From: Duncan Sands <baldrick@free.fr>
Date: Fri Jan 13 10:07:08 2006 +0100
Subject: [PATCH] USBATM: return correct error code when out of memory
Patch-mainline: 2.6.16-rc3

We weren't always returning -ENOMEM.

Signed-off-by: Duncan Sands <baldrick@free.fr>
Signed-off-by: Greg Kroah-Hartman <gregkh@suse.de>

---

 drivers/usb/atm/usbatm.c |    2 ++
 1 files changed, 2 insertions(+), 0 deletions(-)

ff5aeae931ae3874f0a14dd0f1bbab16895c795e
diff --git a/drivers/usb/atm/usbatm.c b/drivers/usb/atm/usbatm.c
index e660a1e..103764d 100644
--- a/drivers/usb/atm/usbatm.c
+++ b/drivers/usb/atm/usbatm.c
@@ -1081,6 +1081,7 @@ int usbatm_usb_probe(struct usb_interfac
 		urb = usb_alloc_urb(iso_packets, GFP_KERNEL);
 		if (!urb) {
 			dev_err(dev, "%s: no memory for urb %d!\n", __func__, i);
+			error = -ENOMEM;
 			goto fail_unbind;
 		}
 
@@ -1090,6 +1091,7 @@ int usbatm_usb_probe(struct usb_interfac
 		buffer = kzalloc(channel->buf_size, GFP_KERNEL);
 		if (!buffer) {
 			dev_err(dev, "%s: no memory for buffer %d!\n", __func__, i);
+			error = -ENOMEM;
 			goto fail_unbind;
 		}
 
-- 
1.1.4.gdcc6

